zerver/lib/streams.py

```diff
@@ -467,7 +467,7 @@ def access_stream_for_delete_or_update(
 def check_basic_stream_access(
     user_profile: UserProfile,
     stream: Stream,
-    sub: Subscription | None,
+    is_subscribed: bool,
```

Comment:
Making these last two keyword-only parameter is probably a good idea.

Generally a coding style principle we have is that it should usually be possible to understand what a function call does by reading the caller, without going to the function's definition. And a bare `False` can't do that, but `is_subscribed=False` is very easy to understand.

Reply:
I'll do a sweep for that across the PR.

Reply:
The `check_basic_stream_access` return types refactor seems to have gotten squashed into a functional change.